### PR TITLE
fix: fetch clients from sql on the fly

### PR DESCRIPTION
**Pull request by markusahlstrand**
Added the clients to the kysely adapter. Didn't yet remove any files but thought we can do that later in a separate PR if it all works out?
We should probably also add some caching here as it makes a couple of queries, but that's for later.

### CODE REVIEW

File: migrate/planetscale.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { BunSqliteDialect } from "kysely-bun-sqlite";
+// @ts-ignore
```
Comment (danielgent) [inline migrate/planetscale.ts:2]: too complex to have types on this? :smile:

File: migrate/sqlite.ts
Comment (danielgent) [inline migrate/sqlite.ts:1]: this PR definitely has some things outside of the description :smile:

File: src/adapters/kysely/clients/index.ts
Comment (danielgent) [inline src/adapters/kysely/clients/index.ts:1]: this was what I said on Slack and whatever PR

Now what are our integration tests doing? `expect(true).toBe(true)`

I suppose we weren't faithfully testing KV storage anyway, but now `clients` is a distinct entity on the tests

File: test/fixtures/context.ts [hunk 2]
```diff
@@ -136,46 +78,20 @@ export function contextFixture(
     });
   }
 
+  if (clients) {
+    clients.forEach((client) => {
+      data.clients.create!(client);
+    });
+  } else {
+    data.clients.create!(client);
```
Comment (danielgent) [inline test/fixtures/context.ts:86]: ? what does this do?

Comment (danielgent) [reply test/fixtures/context.ts:86]: ahhh, this is for the unit tests. I still stand with it but it bothers me less :joy:

File: test/routes/tsoa/token.spec.ts [hunk 3]
```diff
@@ -298,7 +283,7 @@ describe("token", () => {
       expect(tokenData.iss).toBe("https://auth.example.com/");
       expect(tokenData.scope).toEqual("profile");
       expect(tokenData.azp).toBe(undefined);
-      expect(tokenData.sub).toBe("id");
+      expect(tokenData.sub).toBe("clientId");
```
[the remark this answers is not in the document]
Comment (danielgent) [reply test/routes/tsoa/token.spec.ts:286]: I commented elsewhere. The previous unit test fixtures were odd as they were indexed by an id that didn't have to be the id inside the object!

File: migrate/planetscale.ts (hunk 1 above)
Comment (markusahlstrand) [reply migrate/planetscale.ts:2]: I honestly don't know how to do it. Think it's part of the bun runtime? Could we add a declaration for it?

File: src/adapters/in-memory/clients/createClient.ts
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply src/adapters/in-memory/clients/createClient.ts:1]: Yeah, that would be great. The tricky part is that in order to fetch this dynamically we need to be able to read the other in-memory tables which requires us to restructure it a bit. The question is if it's easier to get rid of the entire in-memory version?

File: src/adapters/kysely/clients/index.ts
Comment (markusahlstrand) [reply src/adapters/kysely/clients/index.ts:1]: Yeah, from a testing perspective it hasn't changed much. The next step would be to get the client from application and tenant data

File: test/fixtures/context.ts (hunk 2 above)
Comment (markusahlstrand) [reply test/fixtures/context.ts:86]: So, if we don't pass a client to the fixture it adds a default one. Maybe the fixture should never default to any data and we explicitly add what we need?

File: test/routes/tsoa/token.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (markusahlstrand) [reply test/routes/tsoa/token.spec.ts:286]: Yeah.. so before we were passing a client that had the id `id`. Now we replaced that fixture with the "standard"-fixture which has the id `clientId`

**Pull request merged**